plugins/CircuitViewer/io/MorphologyLoader.cpp

```diff
@@ -87,10 +89,202 @@ const Property PROP_GEOMETRY_QUALITY = {
     {"Geometry quality"}};
 
 const auto LOADER_NAME = "morphology";
+
+struct SDFData
+{
+    std::vector<SDFGeometry> geometries;
+    std::vector<std::set<size_t>> neighbours;
+    std::vector<size_t> materials;
+    // Bifurcation section ID to geometry ID map.
+    std::unordered_map<uint32_t, size_t> bifurcations;
+    std::unordered_map<uint32_t, std::vector<size_t>> sectionGeometries;
+};
+
+bool _isInMask(const brain::neuron::SectionType type, const size_t mask)
+{
+    switch (type)
+    {
+    case brain::neuron::SectionType::soma:
+        return mask & static_cast<size_t>(MorphologySectionType::soma);
+    case brain::neuron::SectionType::axon:
+        return mask & static_cast<size_t>(MorphologySectionType::axon);
+    case brain::neuron::SectionType::dendrite:
+        return mask & static_cast<size_t>(MorphologySectionType::dendrite);
+    case brain::neuron::SectionType::apicalDendrite:
+        return mask &
+               static_cast<size_t>(MorphologySectionType::apical_dendrite);
+    case brain::neuron::SectionType::undefined:
+    default:
+        return true; // What do we do here?
```

Comment:
return false? throw? true seems wrong, but I would have to follow where this is called from.

Reply:
I was thinking that removing things that we don't know about it's probably not the most desirable behaviour. I was having astrocytes in mind when I wrote this code.